tools/topology/sof-apl-keyword-detect.m4

```diff
@@ -0,0 +1,102 @@
+#
+# Topology for ApolloLake with direct attach digital microphones array for
+# keyword detection and triggering use case.
+#
+
+# Include topology builder
+include(`utils.m4')
+include(`dai.m4')
+include(`pipeline.m4')
+
+# Include TLV library
+include(`common/tlv.m4')
+
+# Include Token library
+include(`sof/tokens.m4')
+
+# Include Apollolake DSP configuration
+include(`platform/intel/bxt.m4')
+include(`platform/intel/dmic.m4')
+
+DEBUG_START
+
+#
+# Define the pipelines
+#
+# PCM 0 <-------+- KPBM 0 <-- B0 <-- DMIC6 (DMIC01)
+#               |
+# Keyword <-----+
+
+dnl PIPELINE_PCM_ADD(pipeline,
+dnl     pipe id, pcm, max channels, format,
+dnl     frames, deadline, priority, core)
+
+# Passthrough capture pipeline 1 on PCM 0 using max 2 channels.
+# Schedule 320 frames per 1000us deadline on core 0 with priority 0
+PIPELINE_PCM_ADD(sof/pipe-kfbm-capture.m4,
+	1, 0, 2, s32le,
+	320, 1000, 0, 0)
+
+#
+# DAIs configuration
+#
+
+dnl DAI_ADD(pipeline,
+dnl     pipe id, dai type, dai_index, dai_be,
+dnl     buffer, periods, format,
+dnl     frames, deadline, priority, core)
+
+
+# capture DAI is DMIC 0 using 2 periods
+# Buffers use s16le format, with 320 frame per 1000us on core 0 with priority 0
+DAI_ADD(sof/pipe-dai-capture.m4,
+	1, DMIC, 0, NoCodec-6,
+	PIPELINE_SINK_1, 2, s16le,
+	320, 1000, 0, 0)
+
+dnl PCM_CAPTURE_ADD(name, pipeline, capture)
+PCM_CAPTURE_ADD(DMIC01, 1, PIPELINE_PCM_1)
+
+# keyword detector pipe
+dnl PIPELINE_ADD(pipeline,
+dnl     pipe id, max channels, format,
+dnl     frames, deadline, priority, core)
+PIPELINE_ADD(sof/pipe-detect.m4, 2, 2, s16le, 320, 1000, 0, 0, PIPELINE_SCHED_COMP_1)
+
+# Connect pipelines together
+SectionGraph."pipe-sof-apl-keyword-detect" {
+        index "0"
+
+        lines [
+		# keyword detect
+                dapm(PIPELINE_SINK_2, PIPELINE_SOURCE_1)
+        ]
+}
+# Connect pipelines together
+SectionGraph."pipe-sof-apl-keyword-detect" {
+        index "2"
+
+        lines [
+		# keyword detect
+		dapm(PIPELINE_PCM_1, PIPELINE_DETECT_2)
```

Comment:
@keyonjie can we add both these in the same SectionGraph? I added 2 sections because I was trying something silly but I think we can merge it tomake the topology file a bit compact.

Reply:
not sure if put them to same index "0" work.